src/main/c/bwtidxbuild.c

```diff
@@ -0,0 +1,96 @@
+#include <stdio.h>
+#include <string.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <fcntl.h>
+#include <sys/stat.h>
+#include <sys/mman.h>
+//#include <time.h>
+#include <unistd.h>
+#include <zlib.h>
+#include <errno.h>
+
+#include "bwa/kstring.h"
+#include "bwa/bntseq.h"
+#include "bwa/bwt.h"
+#include "bwa/utils.h"
+#include "bwa/rle.h"
+#include "bwa/rope.h"
+
+int bwt_idx_build(const char *fasta, const char *prefix, const char *algo_type_str);
+
+int bwt_idx_build(const char *fasta, const char *prefix, const char *algo_type_str)
```

Comment:
This can easily get out of sync with the bwa-mem library, so it will be good if at some point this could rely on their implementation by updating the Apache branch. Maybe opening a ticket for reminder will be good to track the issue...